### ISSUE TITLE
Make the timeout period of the transactions configurable

### ISSUE THREAD

**Issue by severussundar** [Azure/data-api-builder#1481]
Make the timeout period of the transaction (associated with a graphQL mutation or REST Post, Put, Patch and Delete request) configurable through the CLI. This is proposed to make sure that resources acquired by a single transaction do not become unavailable to other transactions thereby starving them.

**Comment by pingu2k4** [comment]
Hey, would be great to have the option to customize this. (Changing the timeout in the connection string does nothing).

Any update on this would be great!

**Comment by JerryNixon** [comment]
> Hey, would be great to have the option to customize this. (Changing the timeout in the connection string does nothing).
> 
> We currently aim to allow users to bulk-insert data by uploading CSV files. A real example from a client has around 3,000 items in the CSV file, from which our stored procedure creates each item, as well as creating multiple parameter items for each item created (ending in 3000 items, as well as over 50,000 parameters referenced by said items). Its very possible that this example could grow over time also. We are on a low powered SQL database in Azure currently, and can of course boost this, but could see this being a problem in the future regardless.
> 
> Any update on this would be great!

Connection string does nothing?

**Comment by pingu2k4** [comment]
Yes, changing the timeout specified in the connection string to something larger than 30 seconds isn't respected, and there doesn't appear to be any config allowing modification of any sort of override
